### PR TITLE
Naming Convention: Refactors `AddAzureCosmosClient` and `AddKeyedAzureCosmosClient` API Names

### PR DESCRIPTION
**Pull request by kundadebdatta**
- Refactors `AddAzureCosmosClient` and `AddKeyedAzureCosmosClient` API Names
- Use `/` delimiter for cosmos user agent, instead of `|`
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3906)

### CODE REVIEW

File: src/Components/Aspire.Microsoft.Azure.Cosmos/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -10,5 +10,5 @@ Aspire.Microsoft.Azure.Cosmos.MicrosoftAzureCosmosDBSettings.DisableTracing.get
 Aspire.Microsoft.Azure.Cosmos.MicrosoftAzureCosmosDBSettings.DisableTracing.set -> void
 Aspire.Microsoft.Azure.Cosmos.MicrosoftAzureCosmosDBSettings.MicrosoftAzureCosmosDBSettings() -> void
```
Comment (eerhardt) [inline src/Components/Aspire.Microsoft.Azure.Cosmos/PublicAPI.Unshipped.txt:11]:
```suggestion
Aspire.Microsoft.Azure.Cosmos.MicrosoftAzureCosmosDBSettings.MicrosoftAzureCosmosSettings() -> void
```

If we are dropping `DB` in this library, we should do it for all public APIs in this library.

File: src/Components/Aspire.Microsoft.Azure.Cosmos/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -10,5 +10,5 @@ Aspire.Microsoft.Azure.Cosmos.MicrosoftAzureCosmosDBSettings.DisableTracing.get
...
 Microsoft.Extensions.Hosting.AspireMicrosoftAzureCosmosDBExtensions
```
Comment (eerhardt) [inline src/Components/Aspire.Microsoft.Azure.Cosmos/PublicAPI.Unshipped.txt:12]:
```suggestion
Microsoft.Extensions.Hosting.AspireMicrosoftAzureCosmosExtensions
```

Comment (kundadebdatta) [reply src/Components/Aspire.Microsoft.Azure.Cosmos/PublicAPI.Unshipped.txt]: Thanks! This is now covered with the latest commit.

File: src/Components/Aspire.Microsoft.Azure.Cosmos/PublicAPI.Unshipped.txt (hunk 1 above)
Comment (kundadebdatta) [reply src/Components/Aspire.Microsoft.Azure.Cosmos/PublicAPI.Unshipped.txt]: Thanks! This is now covered with the latest commit.

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks!

**Review by kirankumarkolli — APPROVED** [review]

**Review by Pilchie — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]

**Pull request merged**